lib/elastic_record/index/documents.rb

```diff
@@ -27,20 +30,23 @@ def request_more_hits
       end
 
       def request_next_scroll
-        if scroll_id.nil?
-          response = initial_search_response
-        else
-          response = @elastic_index.scroll(scroll_id, keep_alive)
+        response = scroll_ids.any? ? @elastic_index.scroll(scroll_ids.last, keep_alive) : initial_search_response
+
+        if new_scroll_id = response['_scroll_id']
+          @scroll_ids << new_scroll_id
         end
```

Comment:
I think a scroll is returned on every request, even if the same as previous. You may need to use |=

Reply:
A keep_alive-less scroll request won't return one (since it's technically ending the context). At the time I was experimenting with them, but I can put this back how it was.

Reply:
Sure but you aren’t using a keep-aliveless request anymore. I thought that was a failed idea.

Reply:
Yep, not disagreeing, just stating why it was like that.